Note on building in windows given multiple Python installations

coarsedamavand opened the pull request:
Added Note under the 'build the code in the workspace' section: https://docs.ros.org/en/rolling/Installation/Alternatives/Windows-Development-Setup.html#build-the-code-in-the-workspace

Informs users of cmake-args useful in setting specific Python directories for the windows build on a system with multiple Python installations.

coarsedamavand committed:
Added Note to Windows-Development-Setup.rst

Added Note under the 'build the code in the workspace' section: https://docs.ros.org/en/rolling/Installation/Alternatives/Windows-Development-Setup.html#build-the-code-in-the-workspace

Informs users of cmake-args useful in setting specific Python directories for the windows build on a system with multiple Python installations.

clalancette reviewed (changes requested):
Can you try with `--cmake-args -DPython3_EXECUTABLE=C:\path\to\exe`?  That is what we are going for elsewhere, and I think that is enough without all of the extra settings.

coarsedamavand commented:
> Can you try with `--cmake-args -DPython3_EXECUTABLE=C:\path\to\exe`? That is what we are going for elsewhere, and I think that is enough without all of the extra settings.

Not sure what's changed since I ran this last, but neither my commit nor your suggestion complete when building `rosidl_generator_py`, with the wrong python version being found at [line 22 of rosidl_generator_py_generate_interfaces.cmake](https://github.com/ros2/rosidl_python/blob/179f6880764ada50f145c680c12dac2f6b7fe30d/rosidl_generator_py/cmake/rosidl_generator_py_generate_interfaces.cmake#L22): `find_package(Python3 REQUIRED COMPONENTS Interpreter Development NumPy)`

However, building with the following cmake args DOES complete successfully:
`-DPYTHON_EXECUTABLE=C:/Python38/python.exe `
`-DPython3_ROOT_DIR=C:/Python38 `
`-DPython3_FIND_STRATEGY=LOCATION `
`-DPython3_FIND_REGISTRY=NEVER`
I'll check if all 4 of these are necessary.

coarsedamavand commented:
I did indeed need all 4 cmake args. I'm running on Windows 10, with:
Python 3.8.3 installed from chocolatey in the default location, C:/Python38
And Python 3.9.12 from the web via exe installer, and opted to add it to path during installation.

I will update the documentation accordingly in another commit

coarsedamavand committed:
Merge branch 'ros2:rolling' into rolling

coarsedamavand committed:
Update CMake args fo Python specification

Following testing, updated which CMake args are necessary to avoid build errors arising from incorrect python installs being found and used.

wjwwood commented:
from issue triage meeting, closing
